### ISSUE TITLE
details and attributes via product api

### ISSUE THREAD

**Issue by sdleiw** [Sylius/SyliusShopApiPlugin#232]
Hi,

I need the product details and attributes on the product page, and right now could not find them in the product-by-slug API response, is there another API for that? many thanks.

by the way, the locale parameter in the `products-by-slug/{slug}` is optional, I don't know about product detail, but attributes values are locale based, right?

**Comment by pjedrzejewski** [comment]
Please reopen if still a problem. :)

**Comment by 2monkey** [comment]
Yeah i could not get any description or attribute text from the response :(

`/shop-api/products-by-slug/mug-est?channel=US_WEB&locale=en_US`

`{
    "code": "dcbe2b69-c852-38b6-87c6-3b9bb2d993fc",
    "name": "ASUS - X302UV-FN016T",
    "slug": "mug-est",
    "breadcrumb": "mugs/mug-est",
    "averageRating": 0,
    "taxons": {
        "main": "mugs",
        "others": [
            "mugs"
        ]
    },
    "variants": {
        "dcbe2b69-c852-38b6-87c6-3b9bb2d993fc-variant-0": {
            "code": "dcbe2b69-c852-38b6-87c6-3b9bb2d993fc-variant-0",
            "name": "impedit",
            "axis": [
                "mug_type_medium"
            ],
            "nameAxis": {
                "mug_type_medium": "Mug type Medium mug"
            },
            "price": {
                "current": 217,
                "currency": "USD"
            },
            "images": []
        },
        "dcbe2b69-c852-38b6-87c6-3b9bb2d993fc-variant-1": {
            "code": "dcbe2b69-c852-38b6-87c6-3b9bb2d993fc-variant-1",
            "name": "dolores",
            "axis": [
                "mug_type_double"
            ],
            "nameAxis": {
                "mug_type_double": "Mug type Double mug"
            },
            "price": {
                "current": 866,
                "currency": "USD"
            },
            "images": []
        },
        "dcbe2b69-c852-38b6-87c6-3b9bb2d993fc-variant-2": {
            "code": "dcbe2b69-c852-38b6-87c6-3b9bb2d993fc-variant-2",
            "name": "aliquid",
            "axis": [
                "mug_type_monster"
            ],
            "nameAxis": {
                "mug_type_monster": "Mug type Monster mug"
            },
            "price": {
                "current": 803,
                "currency": "USD"
            },
            "images": []
        }
    },
    "attributes": [],
    "associations": [],
    "images": [
        {
            "code": "main",
            "path": "26/3f/ca0fb44aac6f7bae528b2cf6e96f.jpeg"
        },
        {
            "code": "thumbnail",
            "path": "b6/5c/a770ae8e850df8994cc49f42d924.jpeg"
        },
        {
            "code": "2",
            "path": "3e/86/0aa14f86bc643b7fffc53b053e0f.jpeg"
        },
        {
            "code": "3",
            "path": "4a/d7/3958f33211c3ee8f092ddd85dd0b.jpeg"
        }
    ],
    "_links": {
        "self": {
            "href": "/shop-api/products-by-slug/mug-est"
        }
    }
}`

**Comment by lchrusciel** [comment]
Attributes have to be whitelisted https://github.com/Sylius/SyliusShopApiPlugin#attributes

**Comment by 2monkey** [comment]
@lchrusciel  okey cool now i get some attributes (what does this value mean?)

and how i get the description?


```
"attributes": [
        {
            "code": "mug_material",
            "name": "Mug material",
            "value": [
                "1bd90b9b-246c-399d-b046-4975924a2a85"
            ]
        }
    ],
```

**Comment by lchrusciel** [comment]
This should be a value, you have configured in the backend - https://github.com/Sylius/SyliusShopApiPlugin/blob/master/src/Factory/ProductAttributeValueViewFactory.php#L24-L39. In order to add more data here you have to customize `ProductAttributeValueViewFactory`

**Comment by 2monkey** [comment]
Hmm the Demo Shop has already the information for shortdescription and description.
So i think, i dont need to customize this.

> **The API shop interface works exactly the same as our standard, Twig-based shop UI.**

**Comment by lchrusciel** [comment]
By this sentence, I meant, that the checkout and cart logic is same. I didn't implement shop 1:1. If you need more data, you have to do it on your own or contribute to this repo.
